### PR TITLE
Return success for rrdcached pending on empty

### PR DESCRIPTION
**Pull request by stevenh**
If we have no pending entries for a file return success with no entries instead of returning an error.

### CODE REVIEW

**Comment by stevenh** [comment]
This fixes #808

File: src/rrd_daemon.c [hunk 1]
```diff
@@ -1495,7 +1495,7 @@ static int handle_request_pending(HANDLER_PROTO) /* {{{ */
   if (ci == NULL)
   {
     pthread_mutex_unlock(&cache_lock);
-    rc = send_response(sock, RESP_ERR, "%s\n", rrd_strerror(ENOENT));
+    rc = send_response(sock, RESP_OK, "updates pending\n");
     goto done;
   }
```
Comment (oetiker) [inline src/rrd_daemon.c]: wouldn't it make more sense to just execute the following for loop only IF ci != NULL instead of duplicating the code above ?

**Review by oetiker — CHANGES_REQUESTED** [review]

File: src/rrd_daemon.c (hunk 1 above)
Comment (stevenh) [inline src/rrd_daemon.c]: Sure, also cleaned up the rest of the flow as well as the goto isn't needed

File: src/rrd_daemon.c [hunk 2]
```diff
@@ -1485,28 +1485,19 @@ static int handle_request_pending(HANDLER_PROTO) /* {{{ */
...
-  for (size_t i=0; i < ci->values_num; i++)
-    add_response_info(sock, "%s\n", ci->values[i]);
-
-  pthread_mutex_unlock(&cache_lock);
-  rc = send_response(sock, RESP_OK, "updates pending\n");
-done:
   free(file);
-  return rc;
+  pthread_mutex_unlock(&cache_lock);
+  return send_response(sock, RESP_OK, "updates pending\n");
```
Comment (oetiker) [inline src/rrd_daemon.c]: hmmm still not quite sure if this is correct ... if no pending updates (ci == NULL) are found, then the message 'updates pending' seems odd ...

**Review by oetiker — CHANGES_REQUESTED** [review]

File: src/rrd_daemon.c (hunk 2 above)
Comment (stevenh) [inline src/rrd_daemon.c]: The message will actually be:
```
0 updates pending
```
Which IMO makes perfect sense

Comment (oetiker) [inline src/rrd_daemon.c]: oh my bad ... :) send_response counts what has been added .. all well :) ok

**Pull request merged**